plugins/hashsum_download/plugin_tests/hashsum_download_test.py

```diff
@@ -0,0 +1,246 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright 2016 Kitware Inc.
+#
+#  Licensed under the Apache License, Version 2.0 ( the "License" );
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+###############################################################################
+
+import six
+import hashlib
+
+from tests import base
+
+
+def setUpModule():
+    base.enabledPlugins.append('hashsum_download')
+    base.startServer()
+
+
+def tearDownModule():
+    base.stopServer()
+
+
+class HashsumDownloadTest(base.TestCase):
+    def setUp(self):
+        base.TestCase.setUp(self, assetstoreType='filesystem')
+
```

Comment:
Could you add a comment summarizing the organization of the data in this test.

Something like this:

```
# Create two users: leeloo, zorg
# Create hiearchy:
#   folder1
#     |- file1  [readable by ....]
```